src/main/webapp/service-worker.js

```diff
@@ -0,0 +1,56 @@
+const CACHE_VERSION = 1;
+const CURRENT_CACHES = {
+  censusData: 'census-data-cache-v' + CACHE_VERSION,
+};
+
+
+self.addEventListener('activate', function(event) {
```

Comment:
I know essentially nothing about caching, so this may be a very dumb thing to comment on but bear with me, what does it mean to activate? What exactly is being activated? Just trying to figure out where this function is being called.

Reply:
Right after the service worker is registered (called in HTML onload) it's automatically installed and then activated, so the 'activate' event basically happens right after it's registered.

Reply:
Ahh I got it. Thank you for clarifying that for me!